pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,576 @@
 title: Slice grammar
 ---
 
+This page describes the grammatical rules, usage, and restrictions of the Slice language.
+
+## Slice elements
+
+### Slice files
+
+The `SliceFile` grammar rule is the parser's entry point.
+Once a file's contents have been [preprocessed], the resulting text is matched against this rule, in its entirety.
+
+While the contents of a Slice file may reference definitions in other files, each file is parsed separately.
+There is no notion of 'including' the contents of one file in another, and errors in one file do not affect the parsing of other files.
+
+A slice file consists of any number of [file attributes][attribute] and up to one [mode statement][mode-statement].
```

Comment:
```suggestion
A slice file consists of any number of [file attributes][attribute] and at most one [mode statement][mode-statement].
```

Think this might read a bit better.